server/survey/surveyRepository.js

```diff
@@ -57,8 +57,8 @@ const getSurveyById = async (surveyId, draft = false, client = db) =>
     def => dbTransformCallback(def, draft)
   )
 
-const getSurveyByName = async (surveyName, client = db) =>
-  await client.oneOrNone(
+const getSurveysByName = async (surveyName, client = db) =>
+  await client.many(
```

Comment:
can the query return 0 elements? If yes, you cannot use many, it will reject in that case.
You can use any.  http://vitaly-t.github.io/pg-promise/Database.html#many

Reply:
Actually this should use "map" instead...